lib/signore/cli.rb

```diff
@@ -26,6 +26,15 @@ def run(input: $stdin)
 
     private
 
+    def prego
+      retrieve_sig == Signature.new ? no_signatures_info : retrieve_sig
+    end
```

Comment:
This is much better (everything’s local to the `CLI` class), but it’s still a method that sometimes returns a `Signature` and sometimes returns a `String` – the only reason this isn’t a problem in this case is becuse `puts prego` in `#run` above implicitly calls `#to_s` on whatever `#prego` returns. Still, this means that any caller of this method needs to either switch on the type of the returned object (see whether it got a `Signature` or a `String` back) or only call methods that behave similarly for both a `Signature` and a `String` (like `#to_s`).

Additionally, this implementation calls `retrieve_sig` twice, which means running `signore prego` is suddenly almost twice as slow as before.

How about rather than having a `#prego` query method (one that returns something) have a `#prego` command method (one that does something) which combines `#no_signatures_info` and `#retrieve_sig` (and then have `when 'prego' then prego` in `#run`, of course):

```Ruby
    def prego
      sig = repo.find(tags: tags)
      puts case
           when repo.empty? then 'No signatures found.'
           when sig.empty?  then "Sadly no signatures are tagged #{tags}."
           else sig
           end
    end
```

Note that this also makes it explicit that ‘No signatures found.’ is only an empty-repo case. This will also require implementing `Repo#empty?`, `Signature#empty?` and `Tags#to_s`, but looks more readable to me.